### DISCUSSION TITLE
Option to hide Panel when no fields are rendered due to dependsOn or showOnDetail

### DISCUSSION THREAD

**Discussion by rklugman** [laravel/nova-issues#4606] [Ideas & Feature Requests] [answered]
As I can tell, there is no way to hide Panels when no fields are rendered due to false showOnDetail callbacks or false dependsOn callbacks

**Comment by crynobone** [comment, marked as answer]
~~Panel doesn't know fields "dependsOn" state and therefore can't be reactive.~~

Update: If the panel doesn't have any fields it will be hidden automatically.

**Reply by chrillep** [reply]
@crynobone so the feature request is. Can Panel be rewritten so it "knows fields "dependsOn" state and therefore can be reactive."

Story - As a Nova use i want to be able to hide/show panels so i dont have to set conditions on all the fields in the panel

and thanks for a great product! :)

**Reply by davidhemphill** [reply]
@chrillep This is labeled as implemented and should be available now... If the panel doesn't have any fields it will be hidden automatically.

**Reply by chrillep** [reply]
That's great!

So what I need to do is set dependsOn for every field in a panel?

Since panel does not have the dependsOn method?
